repairnator/docker-images/kubernetes/README.md

```diff
@@ -0,0 +1,141 @@
+# Deploy a high avaliable Repairnator in Kubernetes
+
+*On going pull request*
+
+In order to make Repairnator highly available, it might be good to have configurations and tools to deploy Repairnator in Kubernetes, a preliminary idea here:
+
+- A queue system like ActiveMQ to save build ids
+- A K8s cron job running repairnator scaner to get latest build ids and add them into the queue
+- A set of pods running repairnator pipeline which fetches build id from the queue and does the analysis
+
+# How to set up repairnator on K8s
+
+![Design](K8sRepairnatorDesign.jpg)
+
+
+## Setup mongodb and ActiveMQ
+First set up mongodb on kubernetes with a persistent memory volume(PVC)(This mean that even if you erase the deployment and redeploy nothing will be lost, the database would look the same)
+
+Then create the mongodb deployemnt using mongodb.yaml in the k8s-mongodb folder .
+
+* kubectl create -f k8s-mongodb/mongodb.yaml
+
+Try it out
+* kubectl get pods 
+Our mongodb pod should be named like "mongo-controller-XXXXX" then we can get into the pod 
+* kubectl exec -it mongo-controller-XXXXX bash
+* mongo
+
+Now deploy ActiveMQ for effective queue managements of jobs to the pipeline and scanner. First apply the yaml file inside the queue-for-buildids folder
+* kubectl create -f /queue-for-buildids/activemq.yaml
+To access to the web interface
+* kubectl get pods 
+It should look like activemq-XXXXXXX-XXXXX. Then 
+* kubectl exec -it activemq-XXXXXXX-XXXXX bash
+To expose to localhost for webinterface access and also publishing(sending message to queue) outside the cluster. 
+* kubectl port-forward activemq-XXXXXXX-XXXXX 8161:8161 61613:61613
+Access webinterface with "http://localhost:8161/admin/" in your browser. The default username is "admin" and password is also "admin".
+
+To send message to queue use the publisher.py script in queue-for-buildids folder. Syntax
+* python publisher.py -d queue-name message-1 message-2 message3 ... message-N
```

Comment:
If I understand correctly you will make the scanner use this script to publish a message to the queue, but if this is the case, why not extend the scanner class to simply be able to communicate with the queue directly rather than taking this intermediate step?

Reply:
Again this is only needed for deploying on kubernetes and in the normal case of repairnator there are no request of using activeMQ so according to the "minimal changes to the source" requirement it would not be wise for me to make changes in there. Also  if people want to manually test this out the need to manually inject buildid into the queue for testing.

Reply:
Okay, so you did not want to add a new class either? To me that would seem sensible, but I am not the one who gave you the task so not for me to say really.

Reply:
Yea, I got a set of requirements and I am working in accordance to the supervision to avoid messing up. Then again it's logical since it would be wise that the deploying part should be as decoupled as possible to the source so if there are bugs on only my scripts or repairnator's code is needed for checking and we don't need to worry about the other might mess this up.

Reply:
I would claim that a new class would accomplish the same regarding bugs at least, but okay.